utils/generic.py

```diff
@@ -729,10 +775,10 @@ def get_parents(
     # Add second degree relatives for those samples
     # This is needed to distinguish grandparent - child - parent from child - mother, father down the line
     first_degree_samples = hl.literal(set(first_degree_relatives.keys()))
-    second_degree_samples = kin_ht.filter(
-        (first_degree_samples.contains(kin_ht[i_col]) | first_degree_samples.contains(kin_ht[j_col])) &
-        (kin_ht[kin_col] > second_degree_threshold[0]) &
-        (kin_ht[kin_col] < first_degree_threshold[1])
+    second_degree_samples = relatedness_ht.filter(
+        (first_degree_samples.contains(relatedness_ht[i_col]) | first_degree_samples.contains(relatedness_ht[j_col])) &
+        (relatedness_ht[kin_col] > second_degree_threshold[0]) &
+        (relatedness_ht[kin_col] < first_degree_threshold[1])
```

Comment:
maybe add comment to clarify why you're using the first_degree threshold here, it makes sense but it was confusing the first time i saw it